http-client-core/src/main/java/io/micronaut/http/client/exceptions/HttpClientExceptionUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.client.exceptions;
+
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.http.client.HttpClientConfiguration;
+import io.micronaut.http.client.ServiceHttpClientConfiguration;
+
+/**
+ * Utility Class to work with {@link HttpClientException}.
+ * @author Sergio del Amo
+ * @since 4.0.0
+ */
+public final class HttpClientExceptionUtils {
```

Comment:
overkill to introduce a whole new class for one method. Just move the method to the parent abstract class

Reply:
As well as the JDK client, It's also used in 3 classes in the netty client, none of them share a parent class